syntax = "proto3";

service MsgTransfer {

  rpc image_processor (MsgRequest) returns (MsgReply) {}
  rpc get_server_utilization(Server_Utilization_Request) returns (Server_Utilization_Reply) {}
  rpc get_loaded_models_name (Loaded_Model_Name_Request) returns (Loaded_Model_Name_Reply) {}
  rpc load_specified_model (Load_Specified_Model_Request) returns (Load_Specified_Model_Reply) {}

}

message MsgRequest {

  string model = 1;
  string frame = 2;
  string frame_shape = 3;

}

message MsgReply {

  string result = 2;
  string frame_shape = 3;

}

message Server_Utilization_Request {}

message Server_Utilization_Reply {

  float cpu_usage = 1;
  float memory_usage = 2;

}

message  Loaded_Model_Name_Request {}

message Loaded_Model_Name_Reply {

  string loaded_model_name = 1;

}

message Load_Specified_Model_Request {

  string specified_model = 1;

}

message Load_Specified_Model_Reply {}